server/confdata-binlog-replay.cpp

```diff
@@ -114,28 +169,44 @@ class ConfdataBinlogReplayer : vk::binlog::replayer {
 
     // disable the blacklist because we checked the keys during the previous step
     blacklist_enabled_ = false;
+    int ret_code = 0;
     for (int i = 0; i < nrecords; i++) {
       if (index_offset[i] >= 0) {
-        store_element(reinterpret_cast<const entry_type &>(index_binary_data[index_offset[i]]));
+        auto res = store_element(reinterpret_cast<const entry_type &>(index_binary_data[index_offset[i]]));
+        assert(res != OperationStatus::timed_out);
+        if (res == OperationStatus::throttled_out) {
```

Comment:
Probably, instead of doing conditions here, it's easier to check for `MemoryStatus::NORMAL` in the end. (but I'm not sure that calling `current_memory_status()` inside this function is correct, you can skip this comment).

Reply:
fixed

Reply:
I meant the following:
```
    // no changes here
    for (int i = 0; i < nrecords; i++) {
      if (index_offset[i] >= 0) {
        store_element(reinterpret_cast<const entry_type &>(index_binary_data[index_offset[i]]));
      }
    }
    blacklist_enabled_ = true;
    size_hints_.clear();

    // the only change here
    if (current_memory_status() != MemoryStatus::NORMAL) {
      raise_confdata_oom_error("Can't read confdata snapshot on start");
      return -1;
    }
    return 0;
```

Reply:
fixed